fsspec/spec.py

```diff
@@ -1555,3 +1555,57 @@ def __enter__(self):
 
     def __exit__(self, *args):
         self.close()
+
+
+class AbstractArchiveFileSystem(AbstractFileSystem):
```

Comment:
I think this should go into a module of its own, since it isn't part of the "spec", but a useful shared piece of code for the three implementations. Also, it should probably not be in the top-level `__all__` or imported in `fsspec.__init__` at all, to keep the namespace clean. No other file systems are imported there, except the "spec", in order to keep import times short and expose the canonical ways to make filesystems, like `fsspec.filesystem`.

Reply:
Sure. e2e4e24cb now puts the common superclass into `fsspec.archive`.